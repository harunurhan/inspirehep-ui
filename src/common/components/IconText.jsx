import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'antd';


class IconText extends Component {
  render() {
    const { type, text } = this.props;
    return (
      <span>
        <Icon type={type} style={{ marginRight: 8, fontSize: 16 }} />
        {text}
      </span>
    );
  }
}

IconText.propTypes = {
  type: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default IconText;
