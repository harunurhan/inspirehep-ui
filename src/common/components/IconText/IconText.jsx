import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'antd';

import './IconText.scss';

class IconText extends Component {
  render() {
    const { type, text } = this.props;
    return (
      <span className="__IconText__">
        <Icon className="icon" type={type} />
        {text}
      </span>
    );
  }
}

IconText.propTypes = {
  type: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default IconText;
