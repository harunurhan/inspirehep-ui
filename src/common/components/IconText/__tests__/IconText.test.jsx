import React from 'react';
import { shallow } from 'enzyme';

import IconText from '../IconText';

describe('IconText', () => {
  it('renders with all props set', () => {
    const wrapper = shallow((
      <IconText
        type="info"
        text="Test"
      />
    ));
    expect(wrapper).toMatchSnapshot();
  });
});
