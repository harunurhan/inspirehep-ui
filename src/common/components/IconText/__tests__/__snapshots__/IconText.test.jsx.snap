// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconText renders with all props set 1`] = `
<span
  className="__IconText__"
>
  <Icon
    className="icon"
    type="info"
  />
  Test
</span>
`;
